app/views/books/_book_item.html.erb

```diff
@@ -17,17 +17,17 @@
       by <%= truncate(book.author, length:35) || "Unavailable" %>
     </div>
 
-    <% if book.users.distinct.count > 0 %>
+    <% if book.users.length > 0 %>
       <div class ="books-FlexContainer-presenter">Presented by:
-        <% if book.users.distinct.count <= 3 %>
-          <% book.users.distinct.each_with_index do |user, index| %>
+        <% if book.users.length <= 3 %>
+          <% book.users.each_with_index do |user, index| %>
             <%= link_to user.name, books_path(:user_id => user.id), class:"books-FlexContainer-presenterName" %><% if index < book.users.distinct.count-1 %>,<% end %>
```

Comment:
Why was `distinct` removed? I know it's a costly operation, but it's somewhat necessary to avoid repeated names when a book has been presented twice by the same user. The use of `length` is smart, had no idea length and count had different O's.

Reply:
`length` is made with ruby and `count` is a db op. Usually you want `count` so you don't have to place all the objects in memory, but in this case,  I prefer that trade-off since the `count` query will trigger a N+1 problem (you will make `select count(*))  for each user for each book, making thousands of queries on a single request. This narrows done to a handful of queries.

Reply:
I removed the distinct during debugging btw, will add back

Reply:
Okay, mark this issue as resolved when it is done pls

Reply:
Btw `length` is O(1)

Reply:
Ruby stores the length as an integer on a struct (in C)

But you still load every book into memory. If this was a very large app with 12801281 books, that would absolutely destroy the server.